.github/workflows/individual-image-scanner-quay.yaml

```diff
@@ -0,0 +1,159 @@
+---
+name: Fetch security scan results from quay
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  scans:
+    runs-on: ubuntu-latest
+    outputs:
+      as-output: ${{ steps.access-setup-scan.outputs.VULNERABILITIES_EXIST }}
+      cs-output: ${{ steps.cluster-setup-scan.outputs.VULNERABILITIES_EXIST }}
+      gd-output: ${{ steps.gateway-deployment-scan.outputs.VULNERABILITIES_EXIST }}
+      kr-output: ${{ steps.kcp-registrar-scan.outputs.VULNERABILITIES_EXIST }}
+      sc-output: ${{ steps.shellcheck-scan.outputs.VULNERABILITIES_EXIST }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            access-setup:
+            - 'images/access-setup/**'
+            cluster-setup:
+              - 'images/cluster-setup/**'
+            gateway-deployment:
+              - 'images/gateway-deployment/**'
+            kcp-registrar:
+              - 'images/kcp-registrar/**'
+            shellcheck:
+              - 'test/images/shellcheck/**'
+
+      - name: access-setup scans
+        continue-on-error: true
+        id: access-setup-scan
+        if: steps.filter.outputs.access-setup == 'true'
+        run: |
+          ./test/images/clair-scan/scan-image.sh
+          echo "::set-output name=VULNERABILITIES_EXIST::$(./test/images/clair-scan/scan-image.sh | tail -1)"
+        env:
+          AUTH_BEARER_TOKEN: ${{ secrets.AUTH_BEARER_TOKEN }}
+          IMAGE_NAME: access-setup
+
+      - name: cluster-setup scans
```

Comment:
The same step(s) repeats over and over. I would think there's a way to create a generic step. I'm fine if you tackle this as a latter improvement.

Reply:
Yes, I understand there is repetition, but we need to create a new step for every image as we are going to run that step only if the files in the respective image's path is modified. And also, I am passing unique IMAGE_NAME env variable for each step which is translated into a container.